### PR TITLE
Refactor: ESDB Outbox -> persistent subscription

### PR DESCRIPTION
**Pull request by lzukowski**

### CODE REVIEW

File: event_sourcery/interfaces/outbox_storage_strategy.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import abc
 from typing import Iterator, Tuple
+from uuid import UUID
 
 from event_sourcery.dto import RawEvent
 from event_sourcery.types import StreamId
 
-EntryId = int
+EntryId = int | UUID
```
Comment (Enforcer) [inline event_sourcery/interfaces/outbox_storage_strategy.py:8]: Why like this...? Unions are pesky later

File: event_sourcery_esdb/__init__.py [hunk 2]
```diff
@@ -19,5 +21,8 @@ def __init__(self, esdb: EventStoreDBClient) -> None:
         self._configure(storage_strategy=ESDBStorageStrategy(self._client))
 
     def with_outbox(self, filterer: OutboxFiltererStrategy = dummy_filterer) -> Self:
+        temporal_outbox = f"outbox-test-{uuid4().hex}"
```
Comment (Enforcer) [inline event_sourcery_esdb/__init__.py:24]: uhmm did you mean temporary? EDIT: just found out temporal can have similar meaning, nvm - if it's not a typo, let's leave it although it made me wonder at the first sight.

Comment (Enforcer) [inline event_sourcery_esdb/__init__.py:24]: Do we actually want to include `-test-` in outbox name? Shouldn't this piece be moved over to tests?

File: event_sourcery/interfaces/outbox_storage_strategy.py (hunk 1 above)
Comment (lzukowski) [reply event_sourcery/interfaces/outbox_storage_strategy.py]: Yes. I agree with you. I have removed this change.
But, IMO we need to rethink this API. Let's assume that ESDB int is not a valid entry id. Now it's int but we need a session to discuss outbox.

File: event_sourcery_esdb/__init__.py (hunk 2 above)
Comment (lzukowski) [reply event_sourcery_esdb/__init__.py:24]: maybe `tmp_` ? :D

Comment (lzukowski) [reply event_sourcery_esdb/__init__.py:24]: Yes!!! I 100% agree. will fix this. For some reason, I was assuming this is in `conftest`.

File: event_sourcery/event_store.py [hunk 3]
```diff
@@ -27,12 +26,20 @@ def __init__(
...
         self._event_registry = event_registry
 
-    def outbox(self, publisher: Publisher) -> Outbox:
-        return Outbox(
-            storage_strategy=self._outbox_storage_strategy,
-            event_registry=self._event_registry,
-            publisher=publisher,
-        )
+    def run_outbox(
+            self,
```
Comment (Enforcer) [inline event_sourcery/event_store.py]: Double indent? :P Black should reformat this

File: event_sourcery_esdb/outbox.py [hunk 4]
```diff
@@ -1,202 +1,85 @@
...
-        )
-        self._client.append_event(
-            outbox.metadata,
-            StreamState.ANY,
-            TruncateBeforeEvent(position=Position(outbox.version + 1)),
-        )
+            yield event
+        except Exception:
+            logger.exception("Failed to publish message #%d", entry.id)
+            if entry.retry_count and entry.retry_count >= 2:
```
Comment (Enforcer) [inline event_sourcery_esdb/outbox.py:80]: I assume this is an implementation of up to two retries? Is it possible to retrieve these events later?
In the case of SQL-based event store, they are left behind in the outbox table with retries = 0.

**Review by Enforcer — APPROVED** [review]
I like how EventStore's interface looks after the change. I think removing `Outbox` from `__all__` from top-level `__init__` nicely summarises how this refactoring simplifies library by removing items visible from the outside but also emphasizes how important i moving events from one persistence mechanism to another.

File: event_sourcery/event_store.py (hunk 3 above)
Comment (lzukowski) [reply event_sourcery/event_store.py]: I prefer it this way, but I will change it. Sorry

File: event_sourcery_esdb/outbox.py (hunk 4 above)
Comment (lzukowski) [reply event_sourcery_esdb/outbox.py:80]: Yes, it's up to 3 times, and we have tests for this.
Action that I choose is 
`                self.active_subscription.nack(entry.id, action="park")`
And Park events are available in ESDB to check and to retry. You can do this from UI of ESDB

**Pull request merged**